xenia-build

```diff
@@ -1540,6 +1570,47 @@ class TidyCommand(Command):
             print('Tidy completed successfully.')
             return 0
 
+class StubCommand(Command):
+    """'stub' command."""
+
+    def __init__(self, subparsers, *args, **kwargs):
+        super(StubCommand, self).__init__(
+            subparsers,
+            name='stub',
+            help_short='Create new file(s) in the xenia source tree and run premake',
+            *args, **kwargs)
+        self.parser.add_argument(
+            '--file', default=None,
+            help='Generate a source file at the provided location in the source tree')
+        self.parser.add_argument(
+            '--class', default=None,
+            help='Generate a class pair (.cc/.h) at the provided location in the source tree')
+        self.parser.add_argument(
+            '--target_os', default=None,
+            help='Target OS passed to premake, for cross-compilation')
+
+    def execute(self, args, pass_args, cwd):
+        if args['class']:
+            path = args['class']
+
+            status = generate_source_file('{}.h'.format(path))
+            if status > 0:
+                return status
+            status = generate_source_file('{}.cc'.format(path))
+            if status > 0:
+                return status
```

Comment:
Need to do cleanup here if the `.cc` generation fails, it'll leave a stray `.h`.

Reply:
good point, will look into this tomorrow